web-api/src/business/useCaseHelper/acquireLock.ts

```diff
@@ -7,7 +8,7 @@ export const checkLock = async ({
   onLockError,
   options = {},
 }: {
-  applicationContext: IApplicationContext;
+  applicationContext: ServerApplicationContext;
```

Comment:
Is there a rule-of-thumb for when to use IApplicationContext vs. ServerApplicationContext?   Do the references to IApplicationContext not work here after the move from `shared` to `web-api`?

Reply:
My understanding of the rule of thumb is `ServerApplicationContext` is applicable to functions in `web-api/src` and ClientApplicationContext is for functions in `web-client/src`. `IApplicationContext` is for the `shared/src` fuzzy middle ground functions 😄

Reply:
Yes! `ServerApplicationContext` is meant for functions we know are going to be run on the server(interactors, persistence...) and `IApplicationContext` is meant for functions that could be run on the client or on the server. So like Rachel said generally use `IApplicationContext` in shared/ and `ServerApplicationContext` in web-api/. But make sure it really only gets used on the server before using it!